Watchman.Discord/Areas/Protection/Services/UserMessagesCountService.cs

```diff
@@ -21,24 +22,45 @@ public UserMessagesCountService(IQueryBus queryBus)
 
         public int CountMessages(Contexts contexts)
         {
-            if (_serverMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow)
+            if (ShouldReloadCache())
             {
-                _lastUpdated = DateTime.UtcNow;
+                ReloadCache();
+            }
+
+            return GetOneUserFromOneServerCount(contexts.User.Id, contexts.Server.Id);
+        }
 
-                var query = new GetMessagesQuery();
-                var messages = _queryBus.Execute(query).Messages.ToList();
+        private bool ShouldReloadCache()
+        {
+            return _everyServersMessagesQuantity == null || _lastUpdated.AddHours(12) < DateTime.UtcNow;
+        }
 
-                var serversMessages = messages
-                    .GroupBy(x => x.Server.Id)
-                    .ToList();
+        private void ReloadCache()
+        {
+            _lastUpdated = DateTime.UtcNow;
 
-                this._serverMessagesQuantity = serversMessages
-                    .Select(x => new ServerMessagesCount(x.ToList(), x.Key))
-                    .ToList();
-            }
+            var getAllMessages = new GetMessagesQuery(0);
+            var messages = _queryBus.Execute(getAllMessages).Messages.ToList();
+            var groupedServerMessages = GroupMessagesIntoServers(messages);
+            MakeServerMessagesCounts(groupedServerMessages);
+        }
 
-            var userCount = _serverMessagesQuantity?.FirstOrDefault(x => x.ServerId == contexts.Server.Id)?
-                .UsersMessagesQuantity.FirstOrDefault(x => x.UserId == contexts.User.Id);
+        private IEnumerable<IGrouping<ulong, Message>> GroupMessagesIntoServers(List<Message> messages)
```

Comment:
```suggestion
        private IEnumerable<IGrouping<ulong, Message>> GroupMessagesByServers(List<Message> messages)
```

Reply:
nie lepiej zwracać jako słownik <ulong, IEnumerable<Messages>> ?
lub List zamiast IEnumerable, zależy czy chcemy lazy loading, ale pewnie chcemy żeby nie zapychać ramu

Reply:
A co to zmienia?

Reply:
w sumie może zostać jak jest